### DISCUSSION TITLE
Why does it only works for 30 days?

### DISCUSSION THREAD

**Discussion by darxis** [garrynewman/Control4.Jailbreak#30] [Q&A]
Why does this Jailbreak require to generate and replace the Composer Pro certificates every 30days? 
I'm looking for a way to bypass it. Is it some "trial" feature in Composer Pro?

**Comment by freshcodemonger** [comment]
The reason is likely that the default for generating a self signed certificate is 30 days.  You could probably update the command such that the certs work for longer by following the details here:

https://serverfault.com/questions/916740/local-ssl-ca-always-expires-30-days-after-creation

**Reply by cgbale** [reply]
I believe the reason is that Composer expires the certificate if it doesn’t successfully check-in to the home server after 30 days. I don’t think the generated certificates themselves expire after 30 days.

**Reply by freshcodemonger** [reply]
When you say home server are you talking about the director install?  or c4 corporate?  

Looks to me like the openssl req command is missing the days argument:

https://github.com/garrynewman/Control4.Jailbreak/blob/221d7097d33753db2e0b608c105ca4afe5710d00/UI/Composer.cs#L129

**Reply by cgbale** [reply]
Yes, the C4 cloud checkin is what I was referencing. But I’m not positive on that. 

If it’s just a simple change to the openssl command , that would be great.

**Reply by DanITman** [reply]
The reason is not the certs expiration. The reason is composer has a hard coded check to see when the cert was generated. If it's more than 40 days it doesn't trust it. It's why you have to recreate it.

**Reply by freshcodemonger** [reply]
This spawns a couple questions.  If the cert expires in 30 days (default for creation command as above) does it still work or you have to regen because it expired?  If I try in 35 days will it fail because it expired but it would fail in 40 days anyway since c4 is going to not like it?  I seem to think there is a way to import a CA into your keystore (maybe on the director side) such that if you generated against it (as is happening here) then it is as good as if you had paid a CA to issue you a key/cert that is trusted from a root CA issuer.

**Reply by DanITman** [reply]
The cert does not expire in 30 days. I believe the one Garry creates expires in 365 days. The expiration date doesn't matter. Composer is hard coded to look at the creation date and compare that to the current date.

**Reply by freshcodemonger** [reply]
Looks like it expires in 10 years.

```$openssl x509 -enddate -noout -in cacert-jailbreak.pem```
```notAfter=Mar  5 18:05:57 2034 GMT```

**Reply by freshcodemonger** [reply]
I wonder if you imported the cert into your windows keystore if it would then be ok.  

https://techcommunity.microsoft.com/t5/windows-server-essentials-and/installing-a-self-signed-certificate-as-a-trusted-root-ca-in/ba-p/396105
